### PR TITLE
F/kv scanner support space separator

### PR DESCRIPTION
**Pull request by bodnartibi**
modules/kvformat: support allow-pair-separator-in-values option
    
    We can parse multiple words as values with spaces between them.
    Also handles the case when values are between quotes.
    Leading and trailing spaces will be trimmed from values and keys as well.
    You can put spaces around the separator character.
    
    input:[k=a b c=d]       output:   [k]:[a b]    ; [c]:[d]
    input:[k=a b ,    c=d]  output:   [k]:[a b ,]  ; [c]:[d]
    input:[foo = bar ggg]   output: [foo]:[bar ggg]
    input:[foo = "bar ggg"] output: [foo]:[bar ggg]
    input:[foo = "a=b"]     output: [foo]:[a=b]
    input:[a= b=c]          output:   [a]:[]       ; [b]:[c]
    input:[a=b=c d=e]       output:   [a]:[b=c]    ; [d]:[e]

### CODE REVIEW

**Comment by lbudai** [comment]
Any updates?

**Comment by bodnartibi** [comment]
Final version has been pushed and rebased to master.

File: scripts/style-checker.sh [hunk 1]
```diff
@@ -34,12 +34,13 @@ function setup_root_dir
...
-    astyle --options="$root_dir/.astylerc" --dry-run "$root_dir/*.c" | grep "Formatted" | tee badly-formatted-files.list | wc -l | while read badly_formatted_c_files
+    astyle --options="$root_dir/.astylerc" "$root_dir/*.c" | grep "Formatted" | tee badly-formatted-files.list | wc -l | while read badly_formatted_c_files
     do
         echo "Number of badly formatted files: $badly_formatted_c_files"
         if [ "$badly_formatted_c_files" == "0" ]; then
             exit 0
         else
+            git diff
             cat badly-formatted-files.list
             exit 1
```
Comment (lbudai) [inline scripts/style-checker.sh]: So this patch should not be part of this PR  (I'm not sure that my previous comment is at the right place... )

File: modules/kvformat/linux-audit-scanner.c [hunk 2]
```diff
@@ -134,9 +134,9 @@ _parse_linux_audit_style_hexdump(KVScanner *self)
 }
 
 KVScanner *
-linux_audit_scanner_new(void)
+linux_audit_scanner_new(KVScanner *base)
 {
```
Comment (lbudai) [inline modules/kvformat/linux-audit-scanner.c]: Hmmm.. I'm sure that we've talked about this... this is not a kind of  `_new` function (this is confusing, at least for me). I'll check other changes on tomorrow.

File: modules/kvformat/kv-scanner-generic.c [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+    {
+      _end_next_key(self);
+      self->state = KV_IN_SEPARATOR;
+    }
+  else if (ch == self->super.value_separator)
+    {
+      _end_next_key(self);
+      self->state = KV_KEY_FOUND;
+    }
+  else if (kv_scanner_is_valid_key_character(ch))
```
Comment (lbudai) [inline modules/kvformat/kv-scanner-generic.c]: empty... I'd check it first and when the expr is true I'd return immediately...

File: modules/kvformat/kv-scanner-generic.c [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+
+static inline void
+_in_separator_state(KVScannerGeneric *self, gchar ch)
+{
+  if (ch == '\0')
+    {
+      _extend_value_with_next_key(self);
+      self->state = KV_EOL;
+    }
+  else if (ch == ' ')
```
Comment (lbudai) [inline modules/kvformat/kv-scanner-generic.c]: same pattern here: check if first, when true, return immediately

File: modules/kvformat/kv-scanner-simple.c [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+_extract_value(KVScannerSimple *self)
+{
+  const gchar *cur;
+
+  g_string_truncate(self->super.value, 0);
+  self->super.value_was_quoted = FALSE;
+  cur = &self->super.input[self->super.input_pos];
+
+  self->quote_state = KV_QUOTE_INITIAL;
+  while (*cur && self->quote_state != KV_QUOTE_FINISH)
```
Comment (lbudai) [inline modules/kvformat/kv-scanner-simple.c]: I think this could be splitted intto smaller functions.

**Comment by lbudai** [comment]
I'll send some refactors to your branch.

**Comment by lbudai** [comment]
@bodnartibi : here are my changes: https://github.com/lbudai/syslog-ng/tree/f/kv-scanner-support-space-separator-with-small-refacts
( I cannot send a PR to your branch until you not rebase your branch to master. )

**Comment by bodnartibi** [comment]
@lbudai : I have rebased my branch, and I have skipped b-kil`s commit ("scripts/style-checker.sh: show git diff") as you said

**Comment by lbudai** [comment]
@bodnartibi : I sent a PR to you branch ;)

**Comment by bodnartibi** [comment]
@lbudai Please send new PR, I revert the merge because of stylechecker errors

**Comment by bodnartibi** [comment]
I have merged @lbudai `s suggestions.
